test/e2e/operator_test.go

```diff
@@ -687,26 +687,17 @@ func TestInternalLoadBalancer(t *testing.T) {
 	if err != nil {
 		t.Fatalf("failed to get infrastructure config: %v", err)
 	}
+	platform := infraConfig.Status.Platform
 
-	annotation := map[string]string{}
-	switch infraConfig.Status.Platform {
-	case configv1.AWSPlatformType:
-		annotation["service.beta.kubernetes.io/aws-load-balancer-proxy-protocol"] = "*"
-	case configv1.AzurePlatformType:
-		annotation["service.beta.kubernetes.io/azure-load-balancer-internal"] = "true"
-	case configv1.BareMetalPlatformType:
-		t.Skip("test skipped on bare metal")
-	case configv1.GCPPlatformType:
-		annotation["cloud.google.com/load-balancer-type"] = "Internal"
-	case configv1.LibvirtPlatformType:
-		t.Skip("test skipped on libvirt")
-	case configv1.OpenStackPlatformType:
-		annotation["service.beta.kubernetes.io/openstack-internal-load-balancer"] = "true"
-	case configv1.NonePlatformType:
-		t.Skip("test skipped on \"none\" platform type")
-	case configv1.VSpherePlatformType:
-		t.Skip("test skipped on vSphere")
+	supportedPlatforms := map[configv1.PlatformType]struct{}{
+		configv1.AWSPlatformType:   {},
+		configv1.AzurePlatformType: {},
```

Comment:
Is there a reason not to use `InternalLBAnnotations` to decide whether the platform should be skipped?

Reply:
A related question is how do we consolidate feature enablement checking generally and in a principled way, here and in the origin conformance tests... using the map here might be slightly better in a vacuum here. Followup?

Reply:
If the test is supposed to define the set of platforms where the feature is expected to work, then it should assert that `annotation` is not nil.  But deferring this matter to a follow-up is fine.